genie-agent/src/main/java/com/netflix/genie/agent/execution/services/JobTerminationListenerService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import javax.validation.constraints.NotBlank;
+
+/**
+ * Register an agent to listen for job kill messages from the server
+ * and respond to them.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+public interface JobTerminationListenerService {
```

Comment:
Hmm. What i had in mind was that on the server side I call the service as `JobTerminationService` and agent side `JobTerminationListenerService` as its listening for a termination. We can move to `AgentJobTerminationService`?

Reply:
What is the difference between a `Termination` and a `Kill`. We should be consistent or we're going to think they have a different purposes when they're part of a single purpose.

Why isn't this just something like `JobKillRpcService`?

Reply:
For agent side - Other services in the same package are named as `Agent*`. I am hinging on naming the service `AgentJobKillRpcService`. The service which will actually do the kill, we can call `JobKillService`

For the server - We can call it `JobKillService`. Since its in a different package its clear from the context that this is server side kill.

Reply:
Notice the convention for other services is:
`AgentFooService` (interface)
`GRpcAgentFooServiceImpl` (gRPC implementation)

Reply:
Why?
Both the gRPC listener and ctrl-c should use that component to shut down a launching or running job.

Reply:
Ok will leave `KillService` around. Will later refactor code to use your component. Currently `KillServiceImpl` is not doing anything